lib/sles4sap/database_hana.pm

```diff
@@ -0,0 +1,198 @@
+# SUSE's openQA tests
+#
+# Copyright 2017-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::database_hana;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use saputils qw(check_crm_output get_primary_node get_failover_node calculate_hana_topology);
+use sles4sap::sapcontrol;
+
+our @EXPORT = qw(
+  hdb_stop
+  wait_for_failed_resources
+  wait_for_takeover
+  register_replica
+  get_node_roles
+  find_hana_resource_name
+);
+
+
+=head1 SYNOPSIS
+
+Package contains functions for interacting with hana database and related actions.
+
+=cut
+
+=head2 hdb_stop
+
+    hdb_stop(instance_id=>'00', [switch_user=>'sidadm', command=>'kill']);
+
+Stop hana database using C<HDB stop> command. Function expects to be executed as sidadm, however you can use B<switch_user>
+to execute command using C<sudo su -> as a different user. The user needs to have correct permissions for performing
+requested action.
+Function waits till all DB processes are stopped.
+
+=over
+
+=item * B<instance_id>: Database instance ID. Mandatory.
+
+=item * B<switch_user>: Execute command as specified user with help of C<sudo su ->. Default: undef
+
+=item * B<command>: HDB command to trigger. Default: stop
+
+=back
+
+=cut
+
+sub hdb_stop {
+    my (%args) = @_;
+    my $stop_timeout = 600;
+    $args{command} //= 'stop';
+    croak("Command '$args{command}' is not supported.") unless grep(/$args{command}/, ('kill', 'stop'));
+
+    $args{command} = 'kill -x' if ($args{command} eq 'kill');
```

Comment:
As you are editing on the fly it is maybe better to make a local copy of it (`my $cmd;`) not to edit the variable in the caller scope.

Reply:
Fixed